### PR TITLE
feat(API): stripe integration

### PR DESCRIPTION
**Pull request by Aman-zishan**
### Description of change

- stripe integration
- creates a customer when a new entity is registered
- updates a customer email when the entity email is updated
- decimal point validation
- fix for crypto payment decimal point
- validation for wallet auth status when payment method is crypto
- proper status message for the application
- saves stripeId, invoice URL to the respective tables in backend
- stores entityId to stripe account metadata

### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] `yarn lint` passes with this change
- [ ] `yarn clarinet:check` passes with this change
- [ ] `yarn clarinet:tests` passes with this 
- [ ] `yarn unit-tests` passes with this
- [ ] Run `yarn eol`
- [ ] Run `yarn format`
- [ ] Run `yarn cleanup`
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: standalone/portal-api/src/modules/applications/services/buyer-quick-application.service/stripe.service.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      });
+      await this.stripe.invoiceItems.create({
+        customer: customerId,
+        price: priceId,
+        invoice: invoice.id,
+      });
+      const sendInvoice = await this.stripe.invoices.sendInvoice(invoice.id);
+      return sendInvoice;
+    } catch (error) {
+      console.error('Error creating invoice:', error);
```
Comment (dorucioclea) [inline standalone/portal-api/src/modules/applications/services/buyer-quick-application.service/stripe.service.ts:87]: Please use ILogger, not console.log/console.error. Reason is that in the future we might connect the logger to an aws bucket to save the logs, and the abstraction offers us the possibility to do just that

File: apps/frontend/src/components/modal/invoiceModal.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+};
+function InvoiceModal({ isOpen, onClose }: Props) {
+  const router = useRouter();
+
+  const entityId = router.query.entityId as string;
+  const applicationId = router.query.applicationId as string;
+
+  const handleFinalSubmission = async () => {
+    const application = await applicationService.getApplication(applicationId);
+    console.log(application);
```
Comment (dorucioclea) [inline apps/frontend/src/components/modal/invoiceModal.tsx:26]: Please remove the console log

**Review by dorucioclea — APPROVED** [review]
Requested a few small changes, other than that, looks good to me 🚀

**Comment by Aman-zishan** [comment]
@dorucioclea addressed the reviewed changes

**Comment by dorucioclea** [comment]
Awesome, could you rebase your branch?

**Comment by Aman-zishan** [comment]
> Awesome, could you rebase your branch?

resolved the conflicts by merging main into feat/stripe as rebasing caused some missing changes

**Review by dorucioclea — APPROVED** [review]

**Pull request merged**